MaxText/decode_multi.py

```diff
@@ -0,0 +1,212 @@
+# Copyright 2025 Google LLC
```

Comment:
Consider moving this to MaxText/inference/decode_mutil.py, since this file is inference specific. We can gradually move inference specific things to inference subdir.

Reply:
Should the existing decode.py also be moved then?

Reply:
That's what I was concerned about. So given that does it still make sense to move `decode_multi.py` to the inference folder or keep it in the same space as `decode.py`?

Reply:
I lean toward moving this to inference subdir. But fine if you decide to keep it here

Reply:
Okay to keep as is in this PR and move in a follow up PR.

Reply:
Will add a todo for it and move it in future PR.

Reply:
Went ahead and moved it in this PR since it was simple.